mainwindow.cpp

```diff
@@ -56,12 +56,7 @@ MainWindow::MainWindow(QWidget *parent)
     // Set up the connection to the translator
     connect(translator_.get(), &MarianInterface::pendingChanged, ui_->pendingIndicator, &QProgressBar::setVisible);
     connect(translator_.get(), &MarianInterface::error, this, &MainWindow::popupError);
-    connect(translator_.get(), &MarianInterface::translationReady, this, [&](QString translation) {
-        ui_->outputBox->setText(translation);
-        ui_->localModels->setEnabled(true); // Re-enable model changing
-        ui_->translateAction->setEnabled(true); // Re-enable button after translation is done
-        ui_->translateButton->setEnabled(true);
-    });
+    connect(translator_.get(), &MarianInterface::translationReady, ui_->outputBox, &QTextEdit::setText);
```

Comment:
What happens if you issue multiple translation commands now? Eg you're translating war and peace and it's takign too long and you as the impatient user issue an extra translation request? Would that queue?

Reply:
Yes, but only the last request will be queued.

You can have one translation in progress (that then only lives in the worker thread) and one pending translation (the `MarianInterface::pendingInput_` pointer). While marian is still busy translating, queueing more translations will only replace the `pendingInput_` one.

Reply:
I don't like this because it could accidentally queue an extra long translation that I wouldn't want and there's no way to cancel it. I prefer a new translation to only start once the old one is complete, at least until we can cancel translations?

Reply:
> I don't like this because it could accidentally queue an extra long translation that I wouldn't want and there's no way to cancel it.

You mean something like submitting a form multiple times? That may be better solved by disabling the button as long as the input text remains unchanged. (This would double as an indicator that the translation that pops up after N seconds is actually of the text that you originally typed.)

I imagined the scenario where you pasted in a long email, push "Translate", realise that you forgot to sign it off or something and add that text while it's translating. If the translate button is disabled while translating, you then have to actively wait for it to be enabled again, don't forget to click it, and wait again. If you can always click & queue, you just have to wait once (although for about the same amount of time.)

Reply:
Could we change it back to make sure we  only queue the latest translation.

Reply:
I'm not sure I understand, that's what it does now: when you press "Translate" you replace the queued translation input (or a nullptr if nothing is queued).

Do you mean disable the Translate button while translating is in progress, and only enable it when marian is available again?

Reply:
I've reverted this UI behaviour.